opencensus/stats/exporters/stackdriver_exporter.py

```diff
@@ -0,0 +1,337 @@
+		An error will be raised if there is 
+		already a metric descriptor created with the same name
+		but it has a different aggregation or keys.
+		"""
+		view_measure = view.measure
+		view_aggregation = view.aggregation
+		tag_keys = view.columns
+		view_name = view.name
+
+		metric_type = namespaced_view_name(view_name)
+		value_type = None
+		unit = view_measure.unit
+		# Default metric Kind
+		metric_kind = monitoring_v3.enums.MetricDescriptor.MetricKind.CUMULATIVE
+
+		if view_aggregation.aggregation_type is aggregation.Type.COUNT:
+			value_type = monitoring_v3.enums.MetricDescriptor.ValueType.INT64
+			# If the aggregation type is count, which counts the number of recorded measurements, 
+			# the unit must be "1", because this view does not apply to the recorded values.
+			unit = 1
+		elif view_aggregation.aggregation_type is aggregation.Type.SUM:
+			if view_measure is measure.MeasureInt:
+				value_type = monitoring_v3.enums.MetricDescriptor.ValueType.INT64
+			elif view_measure is measure.MeasureFloat:
+				value_type = monitoring_v3.enums.MetricDescriptor.ValueType.DOUBLE
+		elif view_aggregation.aggregation_type is aggregation.Type.DISTRIBUTION:
+			value_type = monitoring_v3.enums.MetricDescriptor.ValueType.DISTRIBUTION
+		#
+		# Aggreagation type last value is not currently supported by opencesus python stats
+		#
+		# elif view_aggregation.aggregation_type is aggregation.Type.LASTVALUE:
+		#	metric_kind = monitoring_v3.enums.MetricDescriptor.MetricKind.GAUGE
+		# 	if view_measure is measure.MeasureInt:
+		# 		value_type = monitoring_v3.enums.MetricDescriptor.ValueType.INT64
+		# 	elif view_measure is measure.MeasureFloat:
+		# 		value_type = monitoring_v3.enums.MetricDescriptor.ValueType.DOUBLE
+		else:
+			raise Exception("unsupported aggregation type: %s" % str(view_aggregation.aggregation_type))
+
+		display_name_prefix = DEFAULT_DISPLAY_NAME_PREFIX
+		if self.options.metric_prefix != "":
+			display_name_prefix = self.options.metric_prefix
+
+		descriptor = monitoring_v3.types.MetricDescriptor(
+			type = metric_type,
+			metric_kind = metric_kind,
+			value_type = value_type,
+			description = view.description,
+			unit = unit,
+			labels = new_label_descriptors(self.default_labels, view.columns),
+			name = "projects/%s/metricDescriptors/%s" % (self.options.project_id, metric_type),
+			display_name = "%s/%s" % (display_name_prefix, view_name),
+		)
+		project_name = self.client.project_path(self.options.project_id)
+		descriptor = self.client.create_metric_descriptor(project_name, descriptor)
+
+def new_stats_exporter(options):
+	""" new_stats_exporter returns an exporter that uploads stats data to Stackdriver Monitoring.
+	 Only one Stackdriver exporter should be created per ProjectID per process, any subsequent
+	 invocations of NewExporter with the same ProjectID will return an error.
+	"""
+	if str(options.project_id).strip() == "":
+		return None, ERROR_BLANK_PROJECT_ID
```

Comment:
your return pattern is inconsistent, at the end you don't return `None` for error.
and as stated before, this pattern of returning errors isn't the way to go in python, please use exceptions.

Reply:
Thank you for your suggestion. Fixed !